### ISSUE TITLE
https://charts.openshift.io/index.yaml file should return empty YAML rather then 404 error when repo is empty

### ISSUE THREAD

**Issue by pmacik** [openshift-helm-charts/charts#198]
Currently, in case `https://charts.openshift.io` repo is empty, then serving `https://charts.openshift.io/index.yaml` index file returns a `404` error.

The repo should return existing file with a valid empty YAML content such as :
```yaml
[]
```
or
```yaml
--- []
```

Being an empty repository is a corner case but still a valid one IMHO and so should be properly reflected in the repository's index.

Returning 404 error while asking for index in case of the empty repository might cause problems at client side (such as OpenShift Dev Console)

**Comment by pmacik** [comment]
A snippet from OpenShift Console Log:
```
E0614 10:32:34.421669       1 proxy.go:87] Error retrieving index file for &{openshift-helm-charts https://charts.openshift.io false 0x19656e0}: Response for https://charts.openshift.io/index.yaml returned &{404 Not Found 404 HTTP/1.1 1 1 map[Accept-Ranges:[bytes] Access-Control-Allow-Origin:[*] Age:[272] Connection:[keep-alive] Content-Security-Policy:[default-src 'none'; style-src 'unsafe-inline'; img-src data:; connect-src 'self'] Content-Type:[text/html; charset=utf-8] Date:[Mon, 14 Jun 2021 10:32:34 GMT] Etag:[W/"5f9205d6-247b"] Server:[GitHub.com] Vary:[Accept-Encoding] Via:[1.1 varnish] X-Cache:[HIT] X-Cache-Hits:[2] X-Fastly-Request-Id:[759446daf63b44fc6844048254598a32f2d2857c] X-Github-Request-Id:[747C:7985:B1E7C5:F427C1:60C72F32] X-Proxy-Cache:[MISS] X-Served-By:[cache-bwi5036-BWI] X-Timer:[S1623666754.416109,VS0,VE0]] 0xc002bd6020 -1 [] false true map[] 0xc000d34000 0xc000b960b0} with status code 404
E0614 10:32:45.895552       1 proxy.go:87] Error retrieving index file for &{openshift-helm-charts https://charts.openshift.io false 0x19656e0}: Response for https://charts.openshift.io/index.yaml returned &{404 Not Found 404 HTTP/1.1 1 1 map[Accept-Ranges:[bytes] Access-Control-Allow-Origin:[*] Age:[2616] Connection:[keep-alive] Content-Security-Policy:[default-src 'none'; style-src 'unsafe-inline'; img-src data:; connect-src 'self'] Content-Type:[text/html; charset=utf-8] Date:[Mon, 14 Jun 2021 10:32:45 GMT] Etag:[W/"5f74c501-247b"] Server:[GitHub.com] Vary:[Accept-Encoding] Via:[1.1 varnish] X-Cache:[HIT] X-Cache-Hits:[2] X-Fastly-Request-Id:[e99c58e15da6e9fe136d53853748d2efa4d7f439] X-Github-Request-Id:[3DF4:035E:3CA649:7950B1:60C72616] X-Proxy-Cache:[MISS] X-Served-By:[cache-bwi5076-BWI] X-Timer:[S1623666766.890140,VS0,VE0]] 0xc000525280 -1 [] false true map[] 0xc000c06600 0xc001206630} with status code 404
E0614 10:32:47.384978       1 proxy.go:87] Error retrieving index file for &{openshift-helm-charts https://charts.openshift.io false 0x19656e0}: Response for https://charts.openshift.io/index.yaml returned &{404 Not Found 404 HTTP/1.1 1 1 map[Accept-Ranges:[bytes] Access-Control-Allow-Origin:[*] Age:[380] Connection:[keep-alive] Content-Security-Policy:[default-src 'none'; style-src 'unsafe-inline'; img-src data:; connect-src 'self'] Content-Type:[text/html; charset=utf-8] Date:[Mon, 14 Jun 2021 10:32:47 GMT] Etag:[W/"5f74c501-247b"] Server:[GitHub.com] Vary:[Accept-Encoding] Via:[1.1 varnish] X-Cache:[HIT] X-Cache-Hits:[1] X-Fastly-Request-Id:[0ddf86acc605fc3cf86fd4dd7fef45c9ae7917f9] X-Github-Request-Id:[32EA:7C2E:CE1537:1046887:60C72ED3] X-Proxy-Cache:[MISS] X-Served-By:[cache-wdc5560-WDC] X-Timer:[S1623666767.378928,VS0,VE0]] 0xc000524bc0 -1 [] false true map[] 0xc000dcc000 0xc005c3c9a0} with status code 404
E0614 10:32:47.863364       1 proxy.go:87] Error retrieving index file for &{openshift-helm-charts https://charts.openshift.io false 0x19656e0}: Response for https://charts.openshift.io/index.yaml returned &{404 Not Found 404 HTTP/1.1 1 1 map[Accept-Ranges:[bytes] Access-Control-Allow-Origin:[*] Age:[380] Connection:[keep-alive] Content-Security-Policy:[default-src 'none'; style-src 'unsafe-inline'; img-src data:; connect-src 'self'] Content-Type:[text/html; charset=utf-8] Date:[Mon, 14 Jun 2021 10:32:47 GMT] Etag:[W/"5f74c501-247b"] Server:[GitHub.com] Vary:[Accept-Encoding] Via:[1.1 varnish] X-Cache:[HIT] X-Cache-Hits:[1] X-Fastly-Request-Id:[fff2ab1f5edc405bb58b40e7e9f55452cd4e075f] X-Github-Request-Id:[32EA:7C2E:CE1537:1046887:60C72ED3] X-Proxy-Cache:[MISS] X-Served-By:[cache-wdc5579-WDC] X-Timer:[S1623666768.857642,VS0,VE0]] 0xc0005250c0 -1 [] false true map[] 0xc000fc9b00 0xc005c3cf20} with status code 404
E0614 10:32:48.225324       1 proxy.go:87] Error retrieving index file for &{openshift-helm-charts https://charts.openshift.io false 0x19656e0}: Response for https://charts.openshift.io/index.yaml returned &{404 Not Found 404 HTTP/1.1 1 1 map[Accept-Ranges:[bytes] Access-Control-Allow-Origin:[*] Age:[381] Connection:[keep-alive] Content-Security-Policy:[default-src 'none'; style-src 'unsafe-inline'; img-src data:; connect-src 'self'] Content-Type:[text/html; charset=utf-8] Date:[Mon, 14 Jun 2021 10:32:48 GMT] Etag:[W/"5f74c501-247b"] Server:[GitHub.com] Vary:[Accept-Encoding] Via:[1.1 varnish] X-Cache:[HIT] X-Cache-Hits:[1] X-Fastly-Request-Id:[6cd0305c027fea7332072ce96c099c63acd117b7] X-Github-Request-Id:[32EA:7C2E:CE1537:1046887:60C72ED3] X-Proxy-Cache:[MISS] X-Served-By:[cache-wdc5526-WDC] X-Timer:[S1623666768.219243,VS0,VE1]] 0xc0005243e0 -1 [] false true map[] 0xc000d12100 0xc001206420} with status code 404
E0614 10:32:49.191257       1 proxy.go:87] Error retrieving index file for &{openshift-helm-charts https://charts.openshift.io false 0x19656e0}: Response for https://charts.openshift.io/index.yaml returned &{404 Not Found 404 HTTP/1.1 1 1 map[Accept-Ranges:[bytes] Access-Control-Allow-Origin:[*] Age:[287] Connection:[keep-alive] Content-Security-Policy:[default-src 'none'; style-src 'unsafe-inline'; img-src data:; connect-src 'self'] Content-Type:[text/html; charset=utf-8] Date:[Mon, 14 Jun 2021 10:32:49 GMT] Etag:[W/"5f9205d6-247b"] Server:[GitHub.com] Vary:[Accept-Encoding] Via:[1.1 varnish] X-Cache:[HIT] X-Cache-Hits:[1] X-Fastly-Request-Id:[ef144ade32cf3a0689e5496449e874b4b9046ecb] X-Github-Request-Id:[747C:7985:B1E7C5:F427C1:60C72F32] X-Proxy-Cache:[MISS] X-Served-By:[cache-bwi5035-BWI] X-Timer:[S1623666769.181909,VS0,VE1]] 0xc0038f0100 -1 [] false true map[] 0xc000dcc500 0xc001206840} with status code 404
E0614 10:32:51.379953       1 proxy.go:87] Error retrieving index file for &{openshift-helm-charts https://charts.openshift.io false 0x19656e0}: Response for https://charts.openshift.io/index.yaml returned &{404 Not Found 404 HTTP/1.1 1 1 map[Accept-Ranges:[bytes] Access-Control-Allow-Origin:[*] Age:[289] Connection:[keep-alive] Content-Security-Policy:[default-src 'none'; style-src 'unsafe-inline'; img-src data:; connect-src 'self'] Content-Type:[text/html; charset=utf-8] Date:[Mon, 14 Jun 2021 10:32:51 GMT] Etag:[W/"5f9205d6-247b"] Server:[GitHub.com] Vary:[Accept-Encoding] Via:[1.1 varnish] X-Cache:[HIT] X-Cache-Hits:[2] X-Fastly-Request-Id:[3f5f2108f6dff8e098a36ec4a321a985d4b4892a] X-Github-Request-Id:[747C:7985:B1E7C5:F427C1:60C72F32] X-Proxy-Cache:[MISS] X-Served-By:[cache-bwi5083-BWI] X-Timer:[S1623666771.374088,VS0,VE0]] 0xc00091ed60 -1 [] false true map[] 0xc000dcd800 0xc000b966e0} with status code 404
E0614 10:32:52.904169       1 proxy.go:87] Error retrieving index file for &{openshift-helm-charts https://charts.openshift.io false 0x19656e0}: Response for https://charts.openshift.io/index.yaml returned &{404 Not Found 404 HTTP/1.1 1 1 map[Accept-Ranges:[bytes] Access-Control-Allow-Origin:[*] Age:[291] Connection:[keep-alive] Content-Security-Policy:[default-src 'none'; style-src 'unsafe-inline'; img-src data:; connect-src 'self'] Content-Type:[text/html; charset=utf-8] Date:[Mon, 14 Jun 2021 10:32:52 GMT] Etag:[W/"5f9205d6-247b"] Server:[GitHub.com] Vary:[Accept-Encoding] Via:[1.1 varnish] X-Cache:[HIT] X-Cache-Hits:[1] X-Fastly-Request-Id:[2c8477bb71c68b35fa6402a648198dab21244fbc] X-Github-Request-Id:[747C:7985:B1E7C5:F427C1:60C72F32] X-Proxy-Cache:[MISS] X-Served-By:[cache-bwi5058-BWI] X-Timer:[S1623666773.897981,VS0,VE0]] 0xc002cce080 -1 [] false true map[] 0xc000785400 0xc000b96210} with status code 404
E0614 10:32:53.149972       1 proxy.go:87] Error retrieving index file for &{openshift-helm-charts https://charts.openshift.io false 0x19656e0}: Response for https://charts.openshift.io/index.yaml returned &{404 Not Found 404 HTTP/1.1 1 1 map[Accept-Ranges:[bytes] Access-Control-Allow-Origin:[*] Age:[386] Connection:[keep-alive] Content-Security-Policy:[default-src 'none'; style-src 'unsafe-inline'; img-src data:; connect-src 'self'] Content-Type:[text/html; charset=utf-8] Date:[Mon, 14 Jun 2021 10:32:53 GMT] Etag:[W/"5f74c501-247b"] Server:[GitHub.com] Vary:[Accept-Encoding] Via:[1.1 varnish] X-Cache:[HIT] X-Cache-Hits:[1] X-Fastly-Request-Id:[3ebe75ba799f40f0f684cabd4bdde600a3679b6e] X-Github-Request-Id:[32EA:7C2E:CE1537:1046887:60C72ED3] X-Proxy-Cache:[MISS] X-Served-By:[cache-wdc5569-WDC] X-Timer:[S1623666773.143828,VS0,VE1]] 0xc0038f07a0 -1 [] false true map[] 0xc000784c00 0xc001206160} with status code 404
E0614 10:32:53.379730       1 proxy.go:87] Error retrieving index file for &{openshift-helm-charts https://charts.openshift.io false 0x19656e0}: Response for https://charts.openshift.io/index.yaml returned &{404 Not Found 404 HTTP/1.1 1 1 map[Accept-Ranges:[bytes] Access-Control-Allow-Origin:[*] Age:[291] Connection:[keep-alive] Content-Security-Policy:[default-src 'none'; style-src 'unsafe-inline'; img-src data:; connect-src 'self'] Content-Type:[text/html; charset=utf-8] Date:[Mon, 14 Jun 2021 10:32:53 GMT] Etag:[W/"5f9205d6-247b"] Server:[GitHub.com] Vary:[Accept-Encoding] Via:[1.1 varnish] X-Cache:[HIT] X-Cache-Hits:[2] X-Fastly-Request-Id:[b82ca2e59bd213955fed79aef4fa954c35509d74] X-Github-Request-Id:[747C:7985:B1E7C5:F427C1:60C72F32] X-Proxy-Cache:[MISS] X-Served-By:[cache-bwi5026-BWI] X-Timer:[S1623666773.373811,VS0,VE0]] 0xc0038f0860 -1 [] false true map[] 0xc000d13700 0xc0012064d0} with status code 404
E0614 10:32:54.166044       1 proxy.go:87] Error retrieving index file for &{openshift-helm-charts https://charts.openshift.io false 0x19656e0}: Response for https://charts.openshift.io/index.yaml returned &{404 Not Found 404 HTTP/1.1 1 1 map[Accept-Ranges:[bytes] Access-Control-Allow-Origin:[*] Age:[292] Connection:[keep-alive] Content-Security-Policy:[default-src 'none'; style-src 'unsafe-inline'; img-src data:; connect-src 'self'] Content-Type:[text/html; charset=utf-8] Date:[Mon, 14 Jun 2021 10:32:54 GMT] Etag:[W/"5f9205d6-247b"] Server:[GitHub.com] Vary:[Accept-Encoding] Via:[1.1 varnish] X-Cache:[HIT] X-Cache-Hits:[3] X-Fastly-Request-Id:[71d4c71bd16a26bbb7d1cd1d94edaf4b31dcc5a6] X-Github-Request-Id:[747C:7985:B1E7C5:F427C1:60C72F32] X-Proxy-Cache:[MISS] X-Served-By:[cache-bwi5083-BWI] X-Timer:[S1623666774.160183,VS0,VE0]] 0xc00091e020 -1 [] false true map[] 0xc000d12600 0xc005c3c790} with status code 404
E0614 10:33:39.412120       1 proxy.go:87] Error retrieving index file for &{openshift-helm-charts https://charts.openshift.io false 0x19656e0}: Response for https://charts.openshift.io/index.yaml returned &{404 Not Found 404 HTTP/1.1 1 1 map[Accept-Ranges:[bytes] Access-Control-Allow-Origin:[*] Age:[337] Connection:[keep-alive] Content-Security-Policy:[default-src 'none'; style-src 'unsafe-inline'; img-src data:; connect-src 'self'] Content-Type:[text/html; charset=utf-8] Date:[Mon, 14 Jun 2021 10:33:39 GMT] Etag:[W/"5f9205d6-247b"] Server:[GitHub.com] Vary:[Accept-Encoding] Via:[1.1 varnish] X-Cache:[HIT] X-Cache-Hits:[1] X-Fastly-Request-Id:[3b3fe30078adbb974d09a88ff13b59c22042717a] X-Github-Request-Id:[747C:7985:B1E7C5:F427C1:60C72F32] X-Proxy-Cache:[MISS] X-Served-By:[cache-bwi5056-BWI] X-Timer:[S1623666819.406099,VS0,VE1]] 0xc002bd7680 -1 [] false true map[] 0xc000dcdb00 0xc0011aae70} with status code 404
E0614 10:33:41.238085       1 proxy.go:87] Error retrieving index file for &{openshift-helm-charts https://charts.openshift.io false 0x19656e0}: Response for https://charts.openshift.io/index.yaml returned &{404 Not Found 404 HTTP/1.1 1 1 map[Accept-Ranges:[bytes] Access-Control-Allow-Origin:[*] Age:[339] Connection:[keep-alive] Content-Security-Policy:[default-src 'none'; style-src 'unsafe-inline'; img-src data:; connect-src 'self'] Content-Type:[text/html; charset=utf-8] Date:[Mon, 14 Jun 2021 10:33:41 GMT] Etag:[W/"5f9205d6-247b"] Server:[GitHub.com] Vary:[Accept-Encoding] Via:[1.1 varnish] X-Cache:[HIT] X-Cache-Hits:[2] X-Fastly-Request-Id:[e421c0d24971aad011c84bd3ae376cba709648be] X-Github-Request-Id:[747C:7985:B1E7C5:F427C1:60C72F32] X-Proxy-Cache:[MISS] X-Served-By:[cache-bwi5045-BWI] X-Timer:[S1623666821.232096,VS0,VE0]] 0xc003a1c5a0 -1 [] false true map[] 0xc000fb2800 0xc001206b00} with status code 404
E0614 10:33:41.576035       1 proxy.go:87] Error retrieving index file for &{openshift-helm-charts https://charts.openshift.io false 0x19656e0}: Response for https://charts.openshift.io/index.yaml returned &{404 Not Found 404 HTTP/1.1 1 1 map[Accept-Ranges:[bytes] Access-Control-Allow-Origin:[*] Age:[434] Connection:[keep-alive] Content-Security-Policy:[default-src 'none'; style-src 'unsafe-inline'; img-src data:; connect-src 'self'] Content-Type:[text/html; charset=utf-8] Date:[Mon, 14 Jun 2021 10:33:41 GMT] Etag:[W/"5f74c501-247b"] Server:[GitHub.com] Vary:[Accept-Encoding] Via:[1.1 varnish] X-Cache:[HIT] X-Cache-Hits:[1] X-Fastly-Request-Id:[efebaef28e332a07215e94612425f672bcc7b36b] X-Github-Request-Id:[32EA:7C2E:CE1537:1046887:60C72ED3] X-Proxy-Cache:[MISS] X-Served-By:[cache-wdc5551-WDC] X-Timer:[S1623666822.568119,VS0,VE0]] 0xc00091f980 -1 [] false true map[] 0xc000fb3000 0xc000b96630} with status code 404
E0614 10:33:41.817262       1 proxy.go:87] Error retrieving index file for &{openshift-helm-charts https://charts.openshift.io false 0x19656e0}: Response for https://charts.openshift.io/index.yaml returned &{404 Not Found 404 HTTP/1.1 1 1 map[Accept-Ranges:[bytes] Access-Control-Allow-Origin:[*] Age:[434] Connection:[keep-alive] Content-Security-Policy:[default-src 'none'; style-src 'unsafe-inline'; img-src data:; connect-src 'self'] Content-Type:[text/html; charset=utf-8] Date:[Mon, 14 Jun 2021 10:33:41 GMT] Etag:[W/"5f74c501-247b"] Server:[GitHub.com] Vary:[Accept-Encoding] Via:[1.1 varnish] X-Cache:[HIT] X-Cache-Hits:[1] X-Fastly-Request-Id:[d7c02fd48d61678f07bb4c630a70b74b023254a0] X-Github-Request-Id:[32EA:7C2E:CE1537:1046887:60C72ED3] X-Proxy-Cache:[MISS] X-Served-By:[cache-wdc5543-WDC] X-Timer:[S1623666822.811427,VS0,VE0]] 0xc002bd6a60 -1 [] false true map[] 0xc000fb2400 0xc0011aa210} with status code 404
```

**Comment by baijum** [comment]
I have sent one PR to fix this: #199

I tested it like this:

1. Cleaned up the `index.yaml` with the content in #199 . See the commit here: https://github.com/miles2go/repo/commit/40b4e9775cb01ec50d9c0357f0f7d8089bd0acff
2. Created an OWNERS file in the `main` branch. Commit: https://github.com/miles2go/repo/commit/70328d7ba0679636f7629e1c7b33c1358220f0fc
3. Sent a PR: https://github.com/miles2go/repo/pull/117
4. Checked the updated `index.yaml`: https://github.com/miles2go/repo/commit/454160f19aaa7152ea94d9e49cf20812b95b7493
5. Checked the GitHub package release: https://github.com/miles2go/repo/releases/tag/test-org1-psql-service-0.1.32
